src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,170 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        linkLast(value);
```

Comment:
in general linklast and add is the same, maybe it is better to implement logic in add method and then reuse in add(T value, int index). Also please rename linkMiddle in appropriate way